Automation

kalpanaTech opened the pull request:
Hi,
Please find the "Add Languages and Add Skill" automation code for the review. I'm looking forward to some advices to improve my code. I'll appreciate your feedback.
Thanks

kalpanaTech committed:
Login functionality and Language add, update, delete functionalities

kalpanaTech committed:
Skill Add, Update and Delete functionalities

kalpanaTech committed:
User view added Languages and Skills on profile page

kalpanaTech committed:
Added language & skill add cancellation, delete part

kalpanaTech committed:
Add destructive scenarios

kalpanaTech committed:
Final commit

amanbansalnz commented on ProjectMars/Feature/Profile.feature:
```diff
@@ -0,0 +1,299 @@
+﻿Feature: ProfileFeature
+
+This is a platform where users can add and show skills and languages ​​known by users.
```
can you please split your feature files into features,  language and skills! do this with steps definitions too

amanbansalnz commented on ProjectMars/SpecflowPages/Helpers/CommonMethods.cs:
```diff
@@ -0,0 +1,12 @@
+namespace ProjectMars.SpecflowPages.Helpers
+{
+    internal class CommonMethods
+    {
```
no required if its empty

amanbansalnz commented on ProjectMars/SpecflowPages/Helpers/ConstantHelpers.cs:
```diff
@@ -0,0 +1,12 @@
+
+namespace ProjectMars.SpecflowPages.Helpers
+{
+    internal class ConstantHelpers
```
no required if its empty

amanbansalnz commented on ProjectMars/SpecflowPages/Utils/Start.cs:
```diff
@@ -0,0 +1,174 @@
+        
+        public void AddLanguage_Start()
+        {
+            AddLanguage addNewLanguageButtonActionsObj = new AddLanguage();
```
why do you have this?

amanbansalnz replied:
you are missing step definitions please this for specflow exmaple https://docs.specflow.org/projects/specflow/en/latest/ui-automation/Selenium-with-Page-Object-Pattern.html

amanbansalnz commented on ProjectMars/Feature/Profile.feature:
```diff
@@ -0,0 +1,299 @@
+	
+@regression
+Scenario Outline: User add languages
+	When I click on the Add New button
```
i see your do cleanup after each test, but have you cleaned up all data before any test has started?

amanbansalnz reviewed (changes requested):
Hey, I have reviewed your task please see the pr for the suggested changes to make

kalpanaTech committed:
Updated final commit

kalpanaTech commented:
Hi Aman, Thank you for your review. I have updated the code based on your suggestions. Could you please take a look?

I previously added the Start.cs page, thinking it was necessary for SpecFlow, but it wasn't needed, so I removed it along with other unnecessary files.
I added a new step to clean up all data before the tests start.
I combined the steps for adding duplicate data into one, instead of two separate steps.

Thanks
